src/miner.h

```diff
@@ -160,7 +160,7 @@ class BlockAssembler
 public:
     BlockAssembler(const CChainParams& chainparams);
     /** Construct a new block template with coinbase to scriptPubKeyIn */
-    CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn);
+    std::unique_ptr<CBlockTemplate> CreateNewBlock(const CScript& scriptPubKeyIn);
```

Comment:
Shouldn't the caller decide in what kind of pointer to wrap the object?
What if they want to e.g. have a shared pointer instead?